### PR TITLE
ROX-13107: Add missing content sets to the hard coded list for RHCOS 4.7, 4.8 and 4.9.

### PR DESCRIPTION
**Pull request by jvdm**
## Description

Add missing content sets to the hard-coded list for RHCOS 4.7, 4.8, and 4.9.

The additional content sets were found after the following steps, which I would like reviewers to also perform for validity:

1. In the corp network, you can inspect the RHCOS source code to determine which RPM repositories were enabled during the build [on gitlab](https://gitlab.cee.redhat.com/coreos/redhat-coreos/).
2. For each of the branches: 4.7, 4.8 and 4.9:
   1. Open all the `*.repo` files.
   2. For each repository, confirm the content set:
      a. Open https://www.redhat.com/security/data/metrics/repository-to-cpe.json
      b. Replace the architecture with `x86_64`, and grep for the repository path.
      c. Copy the content set name for the entry found.

Notice that for 4.7, `advanced-virt-for-rhel-8-x86_64-eus-rpms` is not used, and the code acts accordingly.

## Tests

- UTs.

### CODE REVIEW

File: pkg/analyzer/detection/detection_test.go [hunk 1]
```diff
@@ -88,8 +88,26 @@ OSTREE_VERSION='49.84.202212201621-0'
...
+				"advanced-virt-for-rhel-8-x86_64-eus-rpms",
+			},
+		},
+		{
+			name: "when ocp/4.7 then should not use advanced-virt",
+			osReleaseContents: `
+ID="rhcos"
+VERSION_ID="4.7"
+OPENSHIFT_VERSION="4.7"
+RHEL_VERSION="8.4"
```
Comment (RTann) [inline pkg/analyzer/detection/detection_test.go:103]: can we add another test to ensure we can correctly get something like `8_DOT_3` or something other than 4 as well?

Comment (jvdm) [reply pkg/analyzer/detection/detection_test.go:103]:
```suggestion
RHEL_VERSION="8.3"
```

File: pkg/analyzer/detection/detection_test.go [hunk 2]
```diff
@@ -88,8 +88,26 @@ OSTREE_VERSION='49.84.202212201621-0'
...
+`,
+			want: []string{
+				"rhel-8-for-x86_64-baseos-eus-rpms__8_DOT_4",
+				"rhel-8-for-x86_64-appstream-eus-rpms__8_DOT_4",
+				"rhel-8-for-x86_64-nfv-tus-rpms__8_DOT_4",
```
Comment (jvdm) [inline pkg/analyzer/detection/detection_test.go:108]:
```suggestion
				"rhel-8-for-x86_64-baseos-eus-rpms__8_DOT_3",
				"rhel-8-for-x86_64-appstream-eus-rpms__8_DOT_3",
				"rhel-8-for-x86_64-nfv-tus-rpms__8_DOT_3",
```

Comment (jvdm) [reply pkg/analyzer/detection/detection_test.go:108]: @ross like this?

File: pkg/analyzer/detection/detection_test.go (hunk 1 above)
Comment (jvdm) [reply pkg/analyzer/detection/detection_test.go:103]: @RTann like this?

Comment (RTann) [reply pkg/analyzer/detection/detection_test.go:103]: yeah as long as we can confirm that it doesn't just happen to work for 8.4 only

**Comment by vikin91** [comment]
Pinging @daynewlee as this PR has influence on [ROX-13801](https://issues.redhat.com/browse/ROX-13801) - we should test the clusters after this is merged and used in stackrox.

**Review by Maddosaurus — APPROVED** [review]
Thanks for the clear instructions!
I was able to verify the path to discover the additional content sets 😄

File: pkg/analyzer/detection/detection.go [hunk 3]
```diff
@@ -145,14 +145,21 @@ func getHardcodedRHCOSContentSets(files analyzer.Files) ([]string, error) {
 	}
 	// Format the content sets based on the metadata found.
 	rhelSuffix := strings.Replace(metadata["RHEL_VERSION"], ".", "_DOT_", 1)
-	return []string{
-		// The mapping from RHCOS version to RHEL minor releases can be found here:
-		// https://access.redhat.com/articles/6907891
```
[the remark this answers is not in the document]
Comment (jvdm) [reply pkg/analyzer/detection/detection.go:150]: Great call. I added the instructions in the JIRA ticket. The JIRA is connected to the PR, which is connected to the git commit that introduced the changes. A README might be better, but we are at least covered.

[the remark this answers is not in the document]
Comment (jvdm) [reply pkg/analyzer/detection/detection.go:150]: By the way, thank you very much for taking the time to double-checking the paths.

**Pull request merged**